tests/phpunit/tests/theme/autoloadThemeMods.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+require_once __DIR__ . '/base.php';
+
+/**
+ * Test autoload for theme mods.
+ *
+ * @package WordPress
+ * @subpackage Theme
+ *
+ * @group themes
+ */
+class Tests_Autoload_Theme_Mods extends WP_Theme_UnitTestCase {
+
+	/**
+	 * Tests that theme mods should not autoloaded after switch_theme.
+	 *
+	 * @ticket 39537
+	 */
+	public function test_that_on_switch_theme_previous_theme_mods_should_not_be_autoload() {
```

Comment:
Related to my comment above, can you also add another unit test which covers switching from/to a child theme?

And maybe even a third test covering switching from a child theme to another child theme with the same parent theme?